HealthGPTUITests/HealthGPTUITests.swift

```diff
@@ -4,12 +4,104 @@
 // SPDX-FileCopyrightText: 2023 Stanford University & Project Contributors
 //
 
+import Security
 import XCTest
+import XCTestExtensions
+import XCTHealthKit
 
 
 final class HealthGPTUITests: XCTestCase {
-    func testExample() throws {
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        resetKeychain()
+
+        try disablePasswordAutofill()
+
+        continueAfterFailure = false
+
+        let app = XCUIApplication()
+        app.launchArguments = ["--showOnboarding"]
+        app.deleteAndLaunch(withSpringboardAppName: "HealthGPT")
+    }
+
+    override func tearDownWithError() throws {
+        resetKeychain()
+        try super.tearDownWithError()
+    }
+
+    func resetKeychain() {
+        let secItemClasses = [
+            kSecClassGenericPassword,
+            kSecClassInternetPassword,
+            kSecClassCertificate,
+            kSecClassKey,
+            kSecClassIdentity
+        ]
+
+        for itemClass in secItemClasses {
+            let spec: [String: Any] = [kSecClass as String: itemClass]
+            SecItemDelete(spec as CFDictionary)
+        }
+    }
```

Comment:
Calling this method in the UI test application only resets key chain entries that are associated with the bundle ID of the UI test application. 
While we could put them in the same keychain group, it would rather be useful to move this method to the testing setup code in the main application target and then trigger the functionality using a feature flag. This would then be executed in the context of the main application and, therefore, should result in the desired outcome.

Please also ensure to credit and cite the code obtained from existing blog posts/GitHub projects in the header of the file or comment in the method to give credit where credit is due.

Reply:
Thank you for the feedback @PSchmiedmayer - I have made these updates. The code was generated by ChatGPT and I have cited it as such.

Reply:
Haha, interesting. ChatGPT probably got if from [here](https://m.pardel.net/resetting-ios-simulator-for-ui-tests-cd7fff57788e), I found this blog post a few months ago when I was looking at a related issue.

But ChatGPT missed all the context to actually put it in the main application target compared to the UI test target. Still requires some developer skills to build software 😄

Reply:
Well, I only prompted it with "Please write a swift method to reset the keychain". So in fairness it didn't have the context needed to know that. :-)